features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumSessionScope.kt

```diff
@@ -211,8 +211,13 @@ internal class RumSessionScope(
         sessionState = if (keepSession) State.TRACKED else State.NOT_TRACKED
         sessionId = UUID.randomUUID().toString()
         sessionStartNs.set(nanoTime)
+        sessionEndedMetricDispatcher.startMetric(
```

Comment:
the question I forgot to ask before: most probably we care only about the sessions which are considered to be tracked? or we want not tracked as well for the metric tracking? I guess it is unlikely

Reply:
Good question, I just checked code of IOS, they will generate nil uuid for the session not sampled so that the metric can be stopped. I just updated the code here when `keepSession` is false, I just skip it.

An integration test is created as well to make sure that  when `sessionSampleRate`=0 , the telemetry event is not received.